### PR TITLE
Custom images for eksd and auto add eksd addons

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

File: build-scripts/addons/repositories.sh [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
 community,https://github.com/canonical/microk8s-community-addons,main
+eksd,https://github.com/canonical/microk8s-aws-addons,main
 "
 
 # List of addon repositories to automatically enable
-ADDONS_REPOS_ENABLED="core"
+ADDONS_REPOS_ENABLED="
+core
+eksd
+"
```
Comment (neoaggelos) [inline build-scripts/addons/repositories.sh:15]: This should be fine
```suggestion
ADDONS_REPOS_ENABLED="core eksd"
```

File: build-scripts/images/parse.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  (cd "${SNAPCRAFT_PART_BUILD}/eks-distro" && git fetch --all && git pull)
+fi
+
+EKS_TRACK=$(echo "${GIT_TAG}" | cut -f1,2 -d'.' | tr -s . - | cut -c2-)
+EKS_RELEASE=$(cat "${SNAPCRAFT_PART_BUILD}"/eks-distro/release/"${EKS_TRACK}"/production/RELEASE)
+
+curl "https://distro.eks.amazonaws.com/kubernetes-$EKS_TRACK/kubernetes-$EKS_TRACK-eks-$EKS_RELEASE.yaml" -o "${SNAPCRAFT_PART_BUILD}/kubernetes-$EKS_TRACK-eks-$EKS_RELEASE.yaml"
+
+mkdir -p $INSTALL/images
+python3 parse.py "${SNAPCRAFT_PART_BUILD}/kubernetes-$EKS_TRACK-eks-$EKS_RELEASE.yaml" "$INSTALL/images"
```
Comment (neoaggelos) [inline build-scripts/images/parse.sh:20]:
```suggestion
python3 parse.py "${SNAPCRAFT_PART_BUILD}/kubernetes-$EKS_TRACK-eks-$EKS_RELEASE.yaml" "$INSTALL/configuration"
```

File: microk8s-resources/wrappers/run-containerd-with-args [hunk 3]
```diff
@@ -48,7 +48,7 @@ fi
 
 app=containerd
 
-
+source "$SNAP/images/images-list"
```
Comment (neoaggelos) [inline microk8s-resources/wrappers/run-containerd-with-args:51]:
```suggestion
source "$SNAP/configuration"
```

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -18,6 +18,7 @@ parts:
     plugin: nil
     override-build: |
       snap install go --classic --channel 1.17/stable
+      python3 -m pip install PyYAML
```
Comment (neoaggelos) [inline snap/snapcraft.yaml:21]: This should probably be `python3-yaml` in the `build-packages` section

**Review by neoaggelos — CHANGES_REQUESTED** [review]
Some comments. 

There is also a `coredns-patched.yaml` which I think is leftover

File: build-scripts/images/parse.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+                for component in manifest["status"]["components"]:
+                    for asset in component["assets"]:
+                        if asset["name"] == "pause-image":
+                            list.write("PAUSE_IMAGE={}\n".format(asset["image"]["uri"]))
+                        elif asset["name"] == "coredns-image":
+                            list.write("COREDNS_IMAGE={}\n".format(asset["image"]["uri"]))
+                        elif asset["name"] == "metrics-server-image":
+                            list.write("METRICS_SERVER_IMAGE={}\n".format(asset["image"]["uri"]))
+        except yaml.YAMLError as exc:
+            print(exc)
```
Comment (neoaggelos) [inline build-scripts/images/parse.py:20]: Adjust the code as you see fit. Reasons for the suggestion:

- Cleanup nested indentation. No need to have nested `with open` blocks, they simply make the code harder to reason about
- sys.argv[2] is already an argument, it should simply the output file.
- load everything in a dict, and simply render the config file in the end.

```suggestion
    eks_manifest_file = sys.argv[1]
    configuration_file = sys.argv[2]

    with open(eks_manifest_file) as fin:
        manifest = yaml.safe_load(fin)

    kvs = {}

    for component in manifest["status"]["components"]:
        for asset in component["assets"]:
            if asset["name"] == "pause-image":
                kvs["PAUSE_IMAGE"] = asset["image"]["uri"]
            elif asset["name"] == "coredns-image":
                kvs["COREDNS_IMAGE"] = asset["image"]["uri"]
            elif asset["name"] == "metrics-server-image":
                kvs["METRICS_SERVER_IMAGE"] = asset["image"]["uri"]

    with open(configuration, "w+") as fout:
        print("\n".join("{}={}".format(k, v) for k, v in kvs.items()))
```

**Comment by ktsakalozos** [comment]
Thank you for the review @neoaggelos. I have addressed your comments.

**Pull request merged**